Extendr fails unit tests on R devel

The test suite does not currently run without error on R devel. See e.g. here:
https://github.com/extendr/extendr/runs/1579358171

This is not related to whether or not libR-sys is compiled with bindgen or uses precompiled bindings.

This is the error:
```
Error: Error in `::`(base, quote) : could not find function "::"
Calls: .OptRequireMethods
Execution halted
error: test failed, to rerun pass '-p extendr-api --lib'
test args::tests::test_args ... 
Error: Process completed with exit code 1.
```
Since it happens when testing the extendr engine, and the engine tests do nothing but start R, we can infer that this happens in the startup code. The fact that the error mentions `.OptRequireMethods` points in the same direction. This function is called at R startup from `setup_Rmainloop()`, here:
https://github.com/wch/r-source/blob/7e99b9a77e4ec0ca3e0ae645f45dcc5717ccf6fb/src/main/main.c#L978-L989

This is the function definition itself:
https://github.com/wch/r-source/blob/d22ee2fc0dc8142b23eed9f46edf76ea9d3ca69a/src/library/profile/Common.R#L88-L96

It can compile with clang, but cannot with GCC. Am I missing something...?

``` console
❯ sudo docker run --rm -it -v "${PWD}:/tmp/test" rocker/r-devel gcc -I/usr/share/R/include -L/usr/lib/R/lib -lR /tmp/test/embed.c
/usr/bin/ld: /tmp/ccJO6TGg.o: in function `main':
embed.c:(.text+0x42): undefined reference to `Rf_initialize_R'
/usr/bin/ld: embed.c:(.text+0x53): undefined reference to `setup_Rmainloop'
collect2: error: ld returned 1 exit status
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -56,7 +56,10 @@ jobs:
 
       - name: Run tests (not Windows)
         if: runner.os != 'Windows'
-        run: cargo test -- --nocapture --test-threads=1
+        run: |
+          export R_HOME=`R -s -e 'cat(R.home())'`
+          export LD_LIBRARY_PATH=${R_HOME}/lib
+          cargo test -- --nocapture --test-threads=1
 
       - name: Build (Windows)
         if: runner.os == 'Windows'
@@ -139,6 +142,8 @@ jobs:
 
       - name: Run tests
         run: |
+          export R_HOME=`R -s -e 'cat(R.home())'`
+          export LD_LIBRARY_PATH=${R_HOME}/lib
           cargo test --manifest-path extendr-engine/Cargo.toml --features libR-sys/use-bindgen -- --nocapture --test-threads=1
           cargo test --manifest-path extendr-macros/Cargo.toml --features libR-sys/use-bindgen -- --nocapture --test-threads=1
           cargo test --manifest-path extendr-api/Cargo.toml --features libR-sys/use-bindgen -- --nocapture --test-threads=1

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -145,5 +145,6 @@ jobs:
           export R_HOME=`R -s -e 'cat(R.home())'`
           export LD_LIBRARY_PATH=${R_HOME}/lib
           cargo test --manifest-path extendr-engine/Cargo.toml --features libR-sys/use-bindgen -- --nocapture --test-threads=1
-          cargo test --manifest-path extendr-macros/Cargo.toml --features libR-sys/use-bindgen -- --nocapture --test-threads=1
           cargo test --manifest-path extendr-api/Cargo.toml --features libR-sys/use-bindgen -- --nocapture --test-threads=1
+          cargo test --manifest-path extendr-macros/Cargo.toml -- --nocapture --test-threads=1
+

--- a/extendr-engine/src/lib.rs
+++ b/extendr-engine/src/lib.rs
@@ -42,11 +42,13 @@ pub fn start_r() {
 
             // In case you are curious.
             // Maybe 8MB is a bit small.
-            // eprintln!("R_CStackLimit={:016x}", R_CStackLimit);
+            println!("R_CStackLimit={:016x}", R_CStackLimit);
 
-            if cfg!(not(target_os = "windows")) {
-                R_CStackLimit = usize::max_value();
-            }
+            //if cfg!(not(target_os = "windows")) {
+            //    R_CStackLimit = usize::max_value();
+            //}
+
+            R_CStackLimit = 100*8388608;
 
             setup_Rmainloop();
         }

--- a/extendr-engine/src/lib.rs
+++ b/extendr-engine/src/lib.rs
@@ -24,12 +24,9 @@ static START_R: Once = Once::new();
 pub fn start_r() {
     START_R.call_once(|| {
         unsafe {
-            // TODO: get the default home dir from libR-sys.
-            if cfg!(unix) {
-                if std::env::var("R_HOME").is_err() {
-                    // env! gets the build-time R_HOME stored by libR-sys
-                    std::env::set_var("R_HOME", env!("R_HOME"));
-                }
+            if std::env::var("R_HOME").is_err() {
+                // env! gets the build-time R_HOME stored by libR-sys
+                std::env::set_var("R_HOME", env!("R_HOME"));
             }
 
             // Due to Rf_initEmbeddedR using __libc_stack_end
@@ -40,7 +37,7 @@ pub fn start_r() {
             // NOTE: R will crash if this is called twice in the same process.
             Rf_initialize_R(
                 3,
-                [cstr_mut!("R"), cstr_mut!("--slave"), cstr_mut!("--no-save")].as_mut_ptr(),
+                [cstr_mut!("R"), cstr_mut!("--slave"), cstr_mut!("--vanilla")].as_mut_ptr(),
             );
 
             // In case you are curious.

--- a/extendr-engine/src/lib.rs
+++ b/extendr-engine/src/lib.rs
@@ -37,18 +37,16 @@ pub fn start_r() {
             // NOTE: R will crash if this is called twice in the same process.
             Rf_initialize_R(
                 3,
-                [cstr_mut!("R"), cstr_mut!("--slave"), cstr_mut!("--vanilla")].as_mut_ptr(),
+                [cstr_mut!("R"), cstr_mut!("--slave"), cstr_mut!("--no-save")].as_mut_ptr(),
             );
 
             // In case you are curious.
             // Maybe 8MB is a bit small.
-            println!("R_CStackLimit={:016x}", R_CStackLimit);
+            // eprintln!("R_CStackLimit={:016x}", R_CStackLimit);
 
-            //if cfg!(not(target_os = "windows")) {
-            //    R_CStackLimit = usize::max_value();
-            //}
-
-            R_CStackLimit = 100*8388608;
+            if cfg!(not(target_os = "windows")) {
+                R_CStackLimit = usize::max_value();
+            }
 
             setup_Rmainloop();
         }

